### PR TITLE
Add post form and comment form

### PR DESCRIPTION
**Pull request by danalvrz**
- Created forms to perform the following functions:

  - Create a Post on behalf of the current_user.
  - Create a comment on behalf of the current_user.
  - Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class CommentsController < ApplicationController
```
Comment (rloterh) [inline app/controllers/comments_controller.rb:1]: - [x]  **New actions is missing** in your comments controller! Kindly ensure that the comments controller have  both _new_ and _create_ actions similar to the one shown below:


```ruby
# controllers/comments_controller.rb

class CommentsController < ApplicationController
  ...

  def new
    @comment = Comment.new
  end

	def create
    @comment = @current_user.comments.new(params)
    @comment.post_id = params[:post_id]

    if @comment.save
      redirect_to post_path(@comment.post)
    else
      @article = Post.find(params[:post_id])
      redirect_to post_path(@post)
    end
  end

  # remind students about the strong parameters in this lesson https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides.
  private 
  def comment_params
    params.require(:comment).permit(:text)
  end

	...
end
```

File: config/routes.rb [hunk 2]
```diff
@@ -1,9 +1,12 @@
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
   resources :users, only: [:index, :show] do
-    resources :posts, only: [:index, :show]
+    resources :posts, only: [:index, :show, :create, :new] do
+      resources :comments, only: [:create]
```
Comment (rloterh) [inline config/routes.rb:5]: - [x] There should be a pair of routes in the _config/routes.rb_ file for creating _comments_. Kindly improve upon this aspect ensuring the _comments_ routes corresponds to:

```ruby
   POST   /posts/:post_id/comments(.:format)     comments#create
 GET   /posts/:post_id/comments/new(.:format)     comments#new
```

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (rloterh) [inline app/controllers/comments_controller.rb:1]: - [x] Kindly ensure that there is a corresponding form in the _views/comments/new.html.erb_ file which accepts input for creating _comments_.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Dante, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦  🚦  

## Good Points 👍  
- Descriptive Pull Request
- Good ReadMe
- No linter errors


## Required Changes ♻️
_Kindly check the comments under the review._

<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 1 more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (danalvrz) [reply app/controllers/comments_controller.rb:1]: done, thanks!

File: config/routes.rb (hunk 2 above)
Comment (danalvrz) [reply config/routes.rb:5]: done, thanks!

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (danalvrz) [reply app/controllers/comments_controller.rb:1]: done, thanks!

Comment (danalvrz) [reply app/controllers/comments_controller.rb:1]: done, thanks!

File: config/routes.rb (hunk 2 above)
Comment (danalvrz) [reply config/routes.rb:5]: done, thanks!

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @danalvrz 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

Great job implementing the changes requested from the previous reviewer.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**